src/util/antlr/ANTLRErrorHandling.cpp

```diff
@@ -27,13 +27,19 @@ ExceptionMetadata generateMetadata(antlr4::ParserRuleContext* ctx) {
 }
 
 // _____________________________________________________________________________
-void ThrowingErrorListener::syntaxError(
-    [[maybe_unused]] antlr4::Recognizer* recognizer,
-    [[maybe_unused]] antlr4::Token* offendingSymbol, size_t line,
-    size_t charPositionInLine, const std::string& msg,
-    [[maybe_unused]] std::exception_ptr e) {
-  throw ParseException{
-      absl::StrCat("Token \"", offendingSymbol->getText(), "\" at line ", line,
-                   ":", charPositionInLine, " ", msg),
-      generateMetadata(recognizer, offendingSymbol, line, charPositionInLine)};
+void ThrowingErrorListener::syntaxError(antlr4::Recognizer* recognizer,
+                                        antlr4::Token* offendingSymbol,
+                                        size_t line, size_t charPositionInLine,
+                                        const std::string& msg,
+                                        [[maybe_unused]] std::exception_ptr e) {
```

Comment:
Why is this argument here if it doesn't appear in the function code?

Reply:
The argument is there because the function is an implementation of a function of the `antlr4::BaseErrorListener` interface. The interface is the way ANTLR provides to handle errors.

Reply:
Yes. This is also visible if you look in the header, where this function in marked `override`.

But it is stil the question: is there any useful content in this `e` argument in the case of a syntax error, or does it always just say `syntax error`.

(This could also (beyond this fix)) be a task for @hannahbast .
Play around with illegal queries and note, how the error messages can be tweaked for increased readability.

Reply:
TL;DR: Not at the moment and probably also not in the future.

The purpose of `std::exception_ptr` is to capture already thrown exceptions in a value. This allows it to be passed around and also rethrown. [Docs](https://en.cppreference.com/w/cpp/error/exception_ptr) I would guess that this allows to do some error handling or clean shutdown and then afterwards still rethrow the exception (`LexerNoViableAltException`, `InputMismatchException`, ...). Since we have our own exception class to throw we don't need this parameter at the moment.